src/SilverStripe/BehatExtension/Context/SilverStripeContext.php

```diff
@@ -497,5 +497,39 @@ public function selectOptionWithJavascript($select, $option) {
 EOS;
 		$this->getSession()->getDriver()->executeScript($script);
 	}
+	
+	
+	/**
+	 * @Override "When /^(?:|I )go to "(?P<page>[^"]+)"$/"
+	 * We override this function to detect issues with .htaccess external redirects.
+	 *
+	 * For instance, if the behat test is being run with a base_url which includes a
+	 * path, e.g. "http://localhost/behat-test-abc123/", .htaccess redirects may take the browser
+	 * to the wrong base path, e.g. "http://localhost/", which will then probably generate
+	 * a apache 404 response, which is pretty standard and we can detect it and give a better
+	 * error message.
+	 *
+	 * @see visitUnsafe - if you want to test an apache 404 response
+	 */
+	public function visit($page){
+		parent::visit($page);
+			
+		// We now check the response body. We would check for the response status code,
+		// but that is not quite possible yet, so this is the best we can do.
+		$page = $this->getSession()->getPage();
+		$title = $page->find('css', 'title')->getHtml(); // getText returns empty string, so have to use getHtml
+		$currentUrl = $this->getSession()->getCurrentUrl();
+		assertNotEquals('404 Not Found', $title, 'A 404 response was detected from the server. If you intended to test an apache 404 response, use visitUnsafe()');
+	}
+	
+	/**
+	 * Like visit, but does not do safety checks.
+	 *
+	 * @Given /^(?:|I )am on "(?P<page>[^"]+)" unsafely$/
+	 * @When /^(?:|I )go to "(?P<page>[^"]+)" unsafely$/
+	 */
+	public function visitUnsafe($page){
+		parent::visit($page);
```

Comment:
I think you can leave out this new method - "unsafely" is a bit confusing, I would've assumed it has something to do with SSL. And I can't really see the use case for it. If you want to write a step which tests that you land on a website's "page not found" screen, that's possible either way, since the title won't be "404 Not Found"